components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/Utils.java

```diff
@@ -148,4 +149,60 @@ public static String getUserIdClaimURI(String federatedIdpName, String tenantDom
         return userIdClaimURI;
     }
 
+    /**
+     * Retrieve the groups claim configured for the federated IDP.
+     *
+     * @param federatedIdpName  Federated IDP name.
+     * @param tenantDomain      Tenant domain.
+     * @return  groups claim configured for the IDP.
+     * @throws IdentityProviderManagementException
+     */
+    public static String getGroupsClaimURIByClaimMappings(String federatedIdpName, String tenantDomain) throws
+            IdentityProviderManagementException {
+
+        String groupsClaimURI = null;
+        ClaimMapping[] claimMappings = getClaimMappings(federatedIdpName, tenantDomain);
+        if (claimMappings == null || claimMappings.length < 1) {
+            return null;
+        }
+
+        /* Here we get the mapping for the application roles claim because federated IDP's groups claim is mapped to
+        local application roles claim.
+         */
+        ClaimMapping groupsClaimMapping = Arrays.stream(claimMappings).filter(claimMapping ->
+                        StringUtils.equals(APP_ROLES_LOCAL_CLAIM, claimMapping.getLocalClaim().getClaimUri()))
+                .findFirst()
+                .orElse(null);
+
+        if (groupsClaimMapping != null) {
+            groupsClaimURI = groupsClaimMapping.getRemoteClaim().getClaimUri();
+        }
+        return groupsClaimURI;
+    }
+
+    /**
+     * Retrieve the claim mappings for the federated IDP.
+     *
+     * @param federatedIdpName  Federated IDP name.
+     * @param tenantDomain      Tenant domain.
+     * @return  groups claim configured for the IDP.
+     * @throws IdentityProviderManagementException
+     */
+    private static ClaimMapping[] getClaimMappings(String federatedIdpName, String tenantDomain) throws
+            IdentityProviderManagementException {
+
+        IdentityProvider idp =
```

Comment:
Do we need to pull the IDP using the IDP service?
Doesn't the context contain information on the IDP and it's claim mappings?

Reply:
Resolved by https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/132/commits/97d487ff047a845a9b889b5d656e09290439ba0f